e2e/infra/istio_setup.go

```diff
@@ -41,27 +54,69 @@ func DeployOperator() (namespace string) {
 	gomega.Expect(projectDir).To(gomega.Not(gomega.BeEmpty()))
 	<-cmd.Execute("oc login -u system:admin").Done()
 
-	namespace, _ = setDockerEnv()
-	// override and use internal address on Deployment
-	err := os.Setenv("IKE_DOCKER_REGISTRY", "172.30.1.1:5000")
-	gomega.Expect(err).To(gomega.Not(gomega.HaveOccurred()))
+	namespace, _ = setDockerEnvForOperatorDeploy()
 
 	<-cmd.ExecuteInDir(projectDir, "make", "deploy-operator").Done()
 	return
 }
 
-func setDockerEnv() (operatorNS, dockerRegistry string) {
-	operatorNS = "istio-workspace-operator"
-	dockerRegistry = "docker-registry-default.127.0.0.1.nip.io:80"
+func setDockerEnvForTestServiceBuild(namespace string) (registry string) {
+	setExampleNamespace(namespace)
+	return setDockerRegistryExternal()
+}
 
-	err := os.Setenv("OPERATOR_NAMESPACE", operatorNS)
+func setDockerEnvForTestServiceDeploy(namespace string) (registry string) {
+	setExampleNamespace(namespace)
+	return setDockerRegistryInternal()
+}
+
+func setDockerEnvForOperatorBuild() (namespace, registry string) {
+	ns := setOperatorNamespace()
+	repo := setDockerRegistryExternal()
+	return ns, repo
+}
+
+func setDockerEnvForOperatorDeploy() (namespace, registry string) {
+	ns := setOperatorNamespace()
+	repo := setDockerRegistryInternal()
+	return ns, repo
+}
+
+func setExampleNamespace(namespace string) {
+	err := os.Setenv("EXAMPLE_NAMESPACE", namespace)
```

Comment:
Rename would imply changing the env var too.

Reply:
https://github.com/Maistra/istio-workspace/pull/134/commits/d7b05a328cea2439ff5e98e1c7a9fdb760e3ccc5